hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/ActionStatusUpdateHandlerService.java

```diff
@@ -0,0 +1,54 @@
+/**
+ * Copyright (c) 2019 Bosch Software Innovations GmbH and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.hawkbit.repository.jpa;
+
+import java.util.List;
+import org.eclipse.hawkbit.repository.ControllerManagement;
+import org.eclipse.hawkbit.repository.EntityFactory;
+import org.eclipse.hawkbit.repository.builder.ActionStatusCreate;
+import org.eclipse.hawkbit.repository.event.remote.ActionStatusUpdateEvent;
+import org.eclipse.hawkbit.repository.model.Action;
+import org.eclipse.hawkbit.repository.model.Action.Status;
+import org.eclipse.hawkbit.security.SystemSecurityContext;
+import org.springframework.context.event.EventListener;
+
+/**
+ * A service that listens and processes the {@link ActionStatusUpdateEvent}
+ *
+ */
+public class ActionStatusUpdateHandlerService {
+
+    private final ControllerManagement controllerManagement;
+    private final EntityFactory entityFactory;
+    private final SystemSecurityContext securityContext;
+
+    ActionStatusUpdateHandlerService(final ControllerManagement controllerManagement, final EntityFactory entityFactory,
+            final SystemSecurityContext securityContext) {
+        this.controllerManagement = controllerManagement;
+        this.entityFactory = entityFactory;
+        this.securityContext = securityContext;
+    }
+
+    @EventListener(classes = ActionStatusUpdateEvent.class)
```

Comment:
I didn't find any place where this event is actually thrown. Have I overlooked something?

Reply:
It is being consumed by an extension project which provides an alternate rollout mechanism through iot-manager. The source code is in iot-sp-extensions/externalRef_for_actions.

Reply:
I see. But this is an entity change event so I'd expect to see this thrown right after the entity is modified. Otherwise you won't cover all code fragments which change the entity. For sure it can be consumed by anybody outside the JPA repository layer (e.g. in your case the extension). But at least somebody has to throw it

Reply:
I used the wrong word. The extention project is indeed throwing the event. So here is the use case for which it is being used:
- The extention project executes the rollout through iot-manager as an alternative for DMF.
- When a rollout is being executed or completed, the iot-manager provides status. 
- This status is resolved by the extension project by determining the appropriate action in the hawkbit.
- It then throws the ActionStatusUpdateEvent which is expected to be handled by hawkbit.
- In future,as an improvement, the AmqpMessageHandlerService can simplify itself by throwing this event. This will allow us to have the update login in one single place.

Reply:
As discussed and agreed, the service has been removed.